[Request]: share_plus-7.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\share\Share.kt:139:55: warning: 'queryIntentActivities(Intent, Int): (Mutable)List<ResolveInfo!>' is deprecated. Deprecated in Java

### Plugin

share_plus-7.2.1

### Use case

C:\Users\zoom\AppData\Local\Pub\Cache\hosted\pub.dev\share_plus-7.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\share\Share.kt:139:55: warning: 'queryIntentActivities(Intent, Int): (Mutable)List<ResolveInfo!>' is deprecated. Deprecated in Java
        val resInfoList = getContext().packageManager.queryIntentActivities(
                                                      ^
C:\Users\zoom\AppData\Local\Pub\Cache\hosted\pub.dev\share_plus-7.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\share\SharePlusPendingIntent.kt:31:20: warning: 'getParcelableExtra(String!): T?' is deprecated. Deprecated in Java
            intent.getParcelableExtra<ComponentName>(Intent.EXTRA_CHOSEN_COMPONENT)


### Proposal

C:\Users\zoom\AppData\Local\Pub\Cache\hosted\pub.dev\share_plus-7.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\share\Share.kt:139:55: warning: 'queryIntentActivities(Intent, Int): (Mutable)List<ResolveInfo!>' is deprecated. Deprecated in Java
        val resInfoList = getContext().packageManager.queryIntentActivities(
                                                      ^
C:\Users\zoom\AppData\Local\Pub\Cache\hosted\pub.dev\share_plus-7.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\share\SharePlusPendingIntent.kt:31:20: warning: 'getParcelableExtra(String!): T?' is deprecated. Deprecated in Java
            intent.getParcelableExtra<ComponentName>(Intent.EXTRA_CHOSEN_COMPONENT)
C:\Users\zoom\AppData\Local\Pub\Cache\hosted\pub.dev\share_plus-7.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\share\Share.kt:139:55: warning: 'queryIntentActivities(Intent, Int): (Mutable)List<ResolveInfo!>' is deprecated. Deprecated in Java
        val resInfoList = getContext().packageManager.queryIntentActivities(
                                                      ^
C:\Users\zoom\AppData\Local\Pub\Cache\hosted\pub.dev\share_plus-7.2.1\android\src\main\kotlin\dev\fluttercommunity\plus\share\SharePlusPendingIntent.kt:31:20: warning: 'getParcelableExtra(String!): T?' is deprecated. Deprecated in Java
            intent.getParcelableExtra<ComponentName>(Intent.EXTRA_CHOSEN_COMPONENT)

--- a/packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/SharePlusPendingIntent.kt
+++ b/packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/SharePlusPendingIntent.kt
@@ -2,6 +2,7 @@ package dev.fluttercommunity.plus.share
 
 import android.content.*
 import android.os.Build
+import androidx.annotation.RequiresApi
 
 /**
  * This helper class allows us to use FLAG_MUTABLE on the PendingIntent used in the Share class,
@@ -21,6 +22,7 @@ internal class SharePlusPendingIntent: BroadcastReceiver() {
     /**
      * Handler called after an action was chosen. Called only on success.
      */
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP_MR1)
     override fun onReceive(context: Context, intent: Intent) {
         // Extract chosen ComponentName
         val chosenComponent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
@@ -29,7 +31,7 @@ internal class SharePlusPendingIntent: BroadcastReceiver() {
         } else {
             // Deprecated in API level 33
             @Suppress("DEPRECATION")
-            intent.getParcelableExtra<ComponentName>(Intent.EXTRA_CHOSEN_COMPONENT)
+            intent.getParcelableExtra(Intent.EXTRA_CHOSEN_COMPONENT)
         }
 
         // Unambiguously identify the chosen action

